projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/SupportedPythonVersions.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.deploy;
+
+import java.util.HashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.beans.factory.annotation.Value;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import lombok.extern.slf4j.Slf4j;
+
+/** Handles operations related to supported python versions for data job deployments. */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class SupportedPythonVersions {
+
+  @Value("#{${datajobs.deployment.supportedPythonVersions}}")
+  private HashMap<String, HashMap<String, String>> supportedPythonVersions;
```

Comment:
I think the requirement to enumerate specific versions supported, is not quite practical. I would imagine rather a condition to narrow those, in case needed. 
May I propose an agile approach towards describing versions compatibility requirements? 
Find https://developer.hashicorp.com/terraform/language/expressions/version-constraints

Reply:
Hi, @ivakoleva 

I think I understand what you mean, but am not sure how practical it is from the administrator point of view. Data Job base images are build for specific major python versions like 3.7, 3.9, etc. Same applies to vdk images, and we need to take into account that users would most likely end up installing packages compatible with specific python versions only.

It seems to me that if we add some looser constraints to the versioning, we may get ourselves in some mess, where we need to do even more work to ensure we don't hit corner cases, deploying data jobs with the wrong images.

If you have some specific suggestion/example in mind, I am open to discuss it.

Reply:
@doks5 I see, what you are concerned with, please correct me if I'm missing something.

From one side, we have the optional constraints on possible base images. Admins may set those rules. For example, python >=3.7 may be a requirement for a VDK deployment in env X.

From the other side, we have an optional per-job configurable python version. It is only valid if satisfying admin constraints. If not customized, will default to latest possible. For example, the image will default to 3.11, that honours the >=3.7 admin constraints. 

https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/job-base-image/Dockerfile-data-job-base#L4 the [parent or base scratch image](https://docs.docker.com/build/building/base-images/) is becoming configurable or not?

Reply:
The base images that can be used would depend on what images are built by the administrators.
The idea is for users to be able to specify what python version they want their jobs to be deployed with, and yes, the base image will be made configurable in a separate PR. The base image will be passed based on the value in supportedPythonVersions.